handler/src/main/java/io/netty/handler/pcap/PcapWriteHandler.java

```diff
@@ -540,6 +541,28 @@ private void completeUDPWrite(InetSocketAddress srcAddr, InetSocketAddress dstAd
         }
     }
 
+    /**
+     * Get the local address of a channel. If the address is a wildcard address ({@code 0.0.0.0} or {@code ::}), and
+     * the address family does not match that of the {@code remote}, return the wildcard address of the {@code remote}'s
+     * family instead.
+     *
+     * @param ch     The channel to get the local address from
+     * @param remote The remote address
+     * @return The fixed local address
+     */
+    private static InetSocketAddress getLocalAddress(Channel ch, InetSocketAddress remote) {
+        InetSocketAddress local = (InetSocketAddress) ch.localAddress();
+        if (remote != null && local.getAddress().isAnyLocalAddress()) {
+            if (local.getAddress() instanceof Inet4Address && remote.getAddress() instanceof Inet6Address) {
+                return new InetSocketAddress("::", local.getPort());
```

Comment:
Consider pre-creating the wildcard `InetAddress` instances and re-use them so we eliminate some overhead.